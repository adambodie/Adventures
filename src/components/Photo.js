import React from 'react';
import ReactTooltip from 'react-tooltip';

const formatDate = date => {
  const monthNames = [ "January", "February", "March", "April", "May", "June", "July", "August", "September", "October","November", "December"];
  const day = date.getDate();
  const monthIndex = date.getMonth();
  const year = date.getFullYear();
  return monthNames[monthIndex] + ' ' + day + ', ' + year;
}

const Photo = props => (
  <div>
        <a data-tip data-for='description'><img src={`https://farm${props.farm}.staticflickr.com/${props.server}/${props.id}_${props.secret}_z.jpg`} alt={props.title}/></a>
        <ReactTooltip place='top' type='light' effect='float' id='description' aria-haspopup='true'>
          <h3>{props.title}</h3>
          <h4>{props.tags}</h4>
          <h4>{formatDate(new Date(props.datetaken))}</h4>
        </ReactTooltip>
        <h6>{props.index} of {props.length}</h6>
  </div>
);

export default Photo;
