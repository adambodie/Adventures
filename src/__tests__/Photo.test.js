import React from "react"
import renderer from "react-test-renderer"
import Photo from "../components/Photo"

describe("Photo", () => {
	it("renders correctly", () => {
		const tree = renderer.create(
			<Photo 
				id={"1"}
				title={"Title"}
				category={"vegas"} 
				index={2}
				length={"45"}
				color={"black"}
				backgroundColor={"blue"} 
				/>
			).toJSON()
		expect(tree).toMatchSnapshot()
	})
})
