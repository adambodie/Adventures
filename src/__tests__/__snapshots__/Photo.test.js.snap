// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Photo renders correctly 1`] = `
<div>
  <img
    alt="Title"
    src="https://adventures.bodiewebdesign.com/photos/images/vegas/1_o.jpg"
  />
  <h3
    style={
      Object {
        "color": "black",
        "textShadow": "-1px 0 blue, 0 1px blue, 1px 0 blue, 0 -1px blue",
      }
    }
  >
    Title
  </h3>
  <h6
    style={
      Object {
        "color": "black",
        "textShadow": "-1px 0 blue, 0 1px blue, 1px 0 blue, 0 -1px blue",
      }
    }
  >
    2
     of 
    45
  </h6>
</div>
`;
